src/summarycode/score.py

```diff
@@ -304,38 +401,57 @@ def check_declared_licenses(license_match_objects):
     """
     return any(is_good_license(license_match_object) for license_match_object in license_match_objects)
 
-
-def get_field_values_from_codebase_resources(
-    codebase,
+def get_field_values_from_resources(
+    resources,
     field_name,
     key_files_only=False,
-    is_string=False
+    is_string=False,
+    is_codebase=False
 ):
     """
     Return a list of values from the `field_name` field of the Resources from
-    `codebase`
-
+    the provided `resources`.
+    
     If `key_files_only` is True, then we only return the field values from
     Resources classified as key files.
 
     If `key_files_only` is False, then we return the field values from Resources
     that are not classified as key files.
     """
     values = []
-    for resource in codebase.walk(topdown=True):
-        if key_files_only:
-            if not resource.is_key_file:
-                continue
-        else:
-            if resource.is_key_file:
-                continue
-        if is_string:
-            value = getattr(resource, field_name, None) or None
-            if value:
-                values.append(value)
-        else:
-            for value in getattr(resource, field_name, []) or []:
-                values.append(value)
+    
+    if is_codebase:
```

Comment:
Let's not make this function complex, we should instead, do a `codebase.get_resource(path=resource_path)` to get the resource objects for a specific path string (make sure you test these for the `strip_root` CLI option).

We should not do two things which are basically the same, depending om a flag value, but we should try to make the inputs same instead, so that same code is valid for both.

Reply:
This is True everywhere else, you are using `resources` to carry both codebase objects and resource dictionaries. This is never a good thing, make sure you only carry a list of resource objects and the codebase object seperately (some cases this is needed for cetain functions) in both cases.

Reply:
By this, I didn't understand why we will be doing the `codebase.get_resource(path=resource_path)`, as we are using a single func `compute_license_score` and inside it we were calling the `get_field_values_from_resources` which can be renamed, as for the `package-summary`, we were passing the Complete Package with all its resources comibined together with `resource field`, shouldn't it be a better approach to use a flag for getting field values for the Package Object that already contains resources with it and normal `field_value` collecttion from the complete codebase for the overall `summary` for the codebase.

 Like using `package_level_summary` flag in the `compute_license_score` func or something else?

Reply:
> understand why we will be doing the codebase.get_resource(path=resource_path)

Because in one case we are using a resource mapping, in another case we are using resource objects, this should not be the case, it should always be obejcts.

See also the following comment:
> you are using resources to carry both codebase objects and resource dictionaries.

This is not okay. We should instead modify and always use a list of resources (for a specific package, or alternatively all the resources in the codebase) passed in the function.

Reply:
Resolved.

Reply:
@AyanSinhaMahapatra Please have a look if this is done correctly this time!